pkg/cmd/server/api/helpers.go

```diff
@@ -17,6 +17,69 @@ import (
 	cmdutil "github.com/openshift/origin/pkg/cmd/util"
 )
 
+var (
+	knownOpenShiftFeatureSet map[string]string
+)
+
+func init() {
+	knownOpenShiftFeatureSet = make(map[string]string, len(KnownOpenShiftFeatures))
+	for _, feature := range KnownOpenShiftFeatures {
+		knownOpenShiftFeatureSet[strings.ToLower(feature)] = feature
+	}
+}
+
+// Add extends feature list with given valid items. They are appended
+// unless already present.
+func (fl *FeatureList) Add(items ...string) error {
+	unknown := []string{}
+	for _, item := range items {
+		if fl.Has(item) {
+			continue
+		}
+		feature, exists := knownOpenShiftFeatureSet[strings.ToLower(item)]
+		if !exists {
+			unknown = append(unknown, item)
+			continue
+		}
+		*fl = append(*fl, feature)
+	}
+	if len(unknown) > 0 {
+		return fmt.Errorf("unknown OpenShift features: %s", strings.Join(unknown, ", "))
+	}
+	return nil
+}
+
+// Delete removes given items from feature list while keeping its original
+// order.
+func (fl *FeatureList) Delete(items ...string) {
+	if len(*fl) < 1 || len(items) < 0 {
```

Comment:
Can you check `== 0`?  It would read easier to my eyes.  Also, I've never seen a slice of `len < 0`, so I suspect easier on your eyes too :)